tests/integration/cocoon/test_cocoon_portfolio_groups.py

```diff
@@ -0,0 +1,324 @@
+                cls.api_factory.build(
+                    lusid.api.TransactionPortfoliosApi
+                ).create_portfolio(
+                    scope="test_scope_20200225",
+                    transaction_portfolio=create_portfolio_model(code),
+                )
+        except lusid.exceptions.ApiException as e:
+            if e.status == 404:
+                print(f"The portfolio {code} already exists")
+
+    def test_multiple_port_groups_multiple_portfolios(self) -> None:
+
+        """
+        Description:
+        ------------
+        Here we test adding multiple new portfolio groups with multiple portfolios.
+        """
+
+        data_frame = pd.read_csv(
+            Path(__file__).parent.joinpath("data/port_group_tests/portfolio_group.csv")
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=data_frame,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={"values.scope": "Scope", "values.code": "FundCode",},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
+
+        self.assertEqual(
+            len(
+                [
+                    port_group
+                    for nested_group in [
+                        port_group.portfolios
+                        for port_group in responses["portfolio_groups"]["success"]
+                    ]
+                    for port_group in nested_group
+                ]
+            ),
+            len(data_frame),
+        )
+
+    def test_one_port_groups_no_portfolios(self) -> None:
+
+        """
+        Description:
+        ------------
+        Here we test adding one new portfolio group with no portfolios.
+        """
+
+        data_frame = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/port_group_tests/portfolio_group_no_portfolio.csv"
+            )
+        )
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=self.scope,
+            data_frame=data_frame,
+            mapping_required={
+                "code": "PortGroupCode",
+                "display_name": "PortGroupDisplayName",
+            },
+            mapping_optional={},
+            file_type="portfolio_group",
+            property_columns=[],
+            properties_scope=None,
+        )
+
+        if responses.get("portfolio_groups").get("errors") != []:
+            for error in responses["portfolio_groups"]["errors"]:
+                print(json.loads(error.body)["title"])
```

Comment:
Better practice here would be to check that the length of the list is greater than 0.

Why did you use .get here "responses.get("portfolio_groups").get("errors")" and then [] syntax a few lines later "responses["portfolio_groups"]["errors"]"

Reply:
Updated to len("response error") > 0.

Re the .get approach - I did this to avoid KeyErrors.
Perhaps a better approach would be:
try foo:
except KeyError:
log (bar)

Reply:
As per above

.get will still throw a key error unless you specify a default, you would need e.g. responses.get("portfolio_groups", {}).get("errors", []). You can see this in that StackOverflow post. 

If the try except block you are referring to is the logging please us an if statement instead as per comment above.